obspy/imaging/tests/test_mopad_script.py

```diff
@@ -45,14 +47,17 @@ def test_script_convert_type_sdr(self):
 Fault plane 1: strike =  77°, dip =  89°, slip-rake = -141°
 Fault plane 2: strike = 346°, dip =  51°, slip-rake =   -1°
 '''
-        expected = expected.encode('utf-8')
 
         result = out.stdout[:-1]
-        if result.startswith(b'b'):
-            # Total mojibake when the system is configured strangely. Mostly
-            # only occurs when building packages, so that LANG=C, and Python
-            # picks ASCII for output encoding.
-            expected = str(expected).encode('ascii')
+        try:
+            expected = expected.encode(sys.stdout.encoding)
+        except:
+            expected = expected.encode('utf-8')
+            if PY3:
+                # This may look like Py3k does extra stuff, but it's actually
+                # saner!
+                expected = repr(expected)
+                expected = expected.encode('ascii')
```

Comment:
`expected` is initially Unicode. If it cannot be converted to stdout encoding, then it's unilaterally converted to a `bytes` containing UTF-8.

Python 2 is "helpful" and converts it back to a regular string.

Python 3, on the other hand, treats `bytes` just like any other object and prints out the `repr`. `repr(bytes)` is a `str`, but stdout is `bytes` (i.e., `fdopen(1, 'wb')`), so it must be encoded again into the stdout encoding. Technically speaking, I probably shouldn't have used ascii here, but the stdout encoding again...